core/management/commands/import_facility_data.py

```diff
@@ -78,16 +77,127 @@ def load_ccn_data_to_facility_model(self, export_path, care_type):
+        measures_per_facility = pd.DataFrame()
+        for measure in individual_measures:
+            measure_values = df[[measure_value_column, "Facility ID"]].loc[df[measure_name_column] == measure]
+            column_map = {}
+            column_map[measure_value_column] = measure
+            measure_values = measure_values.rename(columns=column_map)
+            if measures_per_facility.empty:
+                print("was empty")
+                measures_per_facility = measure_values
+            else:
+                measures_per_facility = pd.merge(measures_per_facility, measure_values, on="Facility ID")
+        measures_per_facility = measures_per_facility.reset_index(drop=True)
+        return measures_per_facility
+        
+    def load_caphs_data(self, export_path, care_type):
+        provider_path = os.path.join(export_path, f"CAHPS - {care_type}.csv")
+        files_with_measures_as_columns = ["Home Health", "Outpatient", "Nursing Homes", "In-Center Hemodialysis"]
+        
+        caphs_df = pd.read_csv(provider_path, low_memory=False, encoding='unicode_escape')
+        
+        if "CMS Certification Number (CCN)" in caphs_df.columns:
+            caphs_df = caphs_df.rename(columns={"CMS Certification Number (CCN)": "Facility ID"})
+
+        
+        if any(file_substring in care_type for file_substring in files_with_measures_as_columns):
+            # filter column
+            measure_columns_by_care_type = {
+                "Home Health" : [
+                    "HHCAHPS Survey Summary Star Rating",
+                ],
+                "Outpatient" : [
+                    "Facilities and staff linear mean score",
+                ],
+                "In-Center Hemodialysis" : [
+                    "Patient Hospital Readmission Category",
+                ],
+                "Nursing Homes" : [
+                    "Overall Rating",
+                ]
+            }   
+            caphs_df = caphs_df[measure_columns_by_care_type[care_type] + ["Facility ID"]]
+        else:
+            caphs_df = self.extract_questions_as_rows(caphs_df, care_type)
+        
+        caphs_df = caphs_df.drop_duplicates()
+        return caphs_df
+        
+    def create_caphs_json_by_row_of_all_caphs_df(self, all_cahps_df):
+        for index, row in all_cahps_df.iterrows():
+            # Convert the row to a dictionary
+            hospital = row.to_dict()  
+            # changing dict to json we need json type to save in the instance
+            hospital = json.dumps(hospital) 
+            if Facility.objects.filter(facility_id=row["Facility ID"]):
+                facility = Facility.objects.filter(facility_id=row["Facility ID"]).first()
+                caphs_metrics = CAPHSMetrics(
+                                    caphs_metric_json=hospital,
+                                    facility=facility
+                                    )
+                caphs_metrics.save()
```

Comment:
Were you able to run this on the whole dataset without performance issues?

Reply:
I was able to run this and didn't have any performance issues.